src/aspire/source/image.py

```diff
@@ -180,6 +181,29 @@ def __getitem__(self, indices):
 
         return IndexedSource(self, indices)
 
+    @property
+    def n(self):
+        return self._n
+
+    @n.setter
```

Comment:
If we do use `IndexSource`, it should should be part of the immutable source conversion, during any non-init metadata change.  I've been trying to hold back on (in case we get Vineet to help with that).

For now I put a guard in that will raise if anything other than this class tries... It's a little ugly though. Can be removed later.

Reply:
Ok. Looks fine for now.